### PR TITLE
Bug fixes

### LINKED ISSUE #84: Fix bugs for Emergency Contact UI
Make emergency contact windows grow and shrink dynamically based on the number of emergency contacts within the window. It is currently hard coded.

### LINKED ISSUE #86: Fix emergency contact UI scroll bar bug
Fixes the scroll bar showing up besides the emergency contact cards when unnecessary.

### LINKED ISSUE #124: Extend name validation to include . and -

### PR DESCRIPTION
**Pull request by kellywsq03**
Closes #124 Closes #86 Closes #84

Unresolved: #123 (to include in UG)

### CODE REVIEW

**Review by Johnwz123 — APPROVED** [review]
LGTM. Just gotta fix the minor merge conflicts.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static Person[] getSamplePersons() {
                                 new Phone("97645132"),
                                 new Relationship("Son"))),
                 new Doctor(new DoctorName("Tan Wei Ming"), new Phone("99119919"), new Email("drtan@gmail.com")),
-                getTagSet("friends")),
+                getTagSet("Mandarin-speaking", "hard of hearing")),
             new Person(new Name("Bernice Yu"), new Phone("99272758"), new Email("berniceyu@example.com"),
```
Comment (aldentantan) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:39]: can change example.com to a different email handle

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public static Person[] getSamplePersons() {
                                 new Relationship("Cousin")
                         )),
                 new Doctor(new DoctorName("Lim Heng Seng"), new Phone("80987123"), new Email("drlim@gmail.com")),
-                getTagSet("colleagues", "friends")),
+                getTagSet("short-term patient")),
             new Person(new Name("Charlotte Oliveiro"), new Phone("93210283"), new Email("charlotte@example.com"),
```
Comment (aldentantan) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:53]: example.com here too

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 3]
```diff
@@ -63,14 +63,14 @@ public static Person[] getSamplePersons() {
                                 new Relationship("Grandson"))
                 ),
                 new Doctor(new DoctorName("Robert Lim"), new Phone("91919191"), new Email("robertlim@hotmail.com")),
-                getTagSet("neighbours")),
+                getTagSet("needs hearing aid")),
             new Person(new Name("David Li"), new Phone("91031282"), new Email("lidavid@example.com"),
```
Comment (aldentantan) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:67]: same as the comments above

**Review by aldentantan — APPROVED** [review]
Looks good to me!

**Pull request merged**